rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/security/ToggleSecurityFilter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.coordinator.security;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.ext.Provider;
+import java.security.Principal;
+
+@PreMatching
+@Provider
+public class ToggleSecurityFilter implements ContainerRequestFilter {
```

Comment:
When I tried, for `Quarkus` this option `quarkus.smallrye-jwt.enabled=true` didn't work for some corner cases.

You are correct that there would be solution switching on/off JWT, but for every different server, user will have to search solution for this problem and he will have to apply it in his project.

Here my motive was to provide a generic/simple solution which would work for all the server.

Reply:
When I set `quarkus.smallrye-jwt.enabled=false`, then if I call any endpoints without token, I would not get any results.

But here we are not just talking about Quarkus, other servers should also be considered.

Reply:
My position was that when I switch on the JWT for the runtime I want to use it in all services. Every runtime - every server - has its own way how to setup JWT. For WildFly it will be most probably some subsystem configuration. For quarkus it's the application property.

But ok, probably being able to configure the JWT per component makes sense.

What does it mean that setting the `quarkus.smallrye-jwt.enabled=false` you get no result? Like the response is lost?
You need to consider that user will set `quarkus.smallrye-jwt.enabled=false` and `io.narayana.lra.security.enabled=true` - the JWT is not enabled by runtime. There has to be expected behaviour. In fact if it returns nothing then it's maybe correct from the JWT spec and then `io.narayana.lra.security.enabled=false` should behave in the same way (if it's something that JWT prescribes). Can you please elaborate a more on this?

Reply:
If we do not set `quarkus.smallrye-jwt.enabled` property then also it works. I have tried this.

Here user should use only `io.narayana.lra.security.enabled` to enable/disable security.